### ISSUE TITLE
Metric to quantify router overhead

### ISSUE THREAD

**Issue by lennyburdette** [apollographql/router#1949]
Customers often ask how much time is pure router overhead. I'm not exactly sure how that would be calculated, as it's not as simple as `(total_time - sum(subgraph_fetch_time))` because fetches can happen in parallel. Is there a way to find the amount of time in a span *not* spent in child spans? 

Essentially, I want the time marked with `*` in this diagram as a Prometheus metric: 

```
 **            *                ***                      ***
|-----------------------------------------------------------|
   |----------|   
                |--------------|
                |----------|
                                   |--------------------| 
```

This would allow graph platform teams to discover Router regressions after upgrades or after changing configuration.

**Comment by garypen** [comment]
I wonder if this is already available from span tag data...

I've often wondered what the `busy_ns` and `idle_ns` tags on a span in jaeger's UI represent. I did some digging around and couldn't find many clues apart from: https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/struct.OpenTelemetryLayer.html#method.with_tracked_inactivity

This is enabled (I think by default) in the router.

So, for a representative request to the router I noted that the request span reported(in ms):
 - total: 160
 - idle: 154
 - busy: 6
 (idle and busy are reported in nanoseconds, so some conversion and rounding was required)

That appears to suggest that the data you are looking for is there. It's not a prometheus metric and I'm not sure how tricky it would be to extract that, but this might be a good starting place.

**Comment by BrynCooke** [comment]
Span metrics are implemented in #1948 
Probably needs some more work, but for now let's flag this as blocked.

**Comment by IceHoppr** [comment]
Yes, we would be interested in this metric about 'overhead' addition by Router. Essentially this is the time spent for processing in request and response paths at Router for a GQL request. This metric will help us in visualize and benchmark Router performance over the upgrades and custom functionality changes.
